go/vt/mysqlctl/mysqld.go

```diff
@@ -688,7 +687,12 @@ func deleteTopDir(dir string) (removalErr error) {
 // executeMysqlCommands executes some SQL commands,
 // using the mysql command line tool.
 func (mysqld *Mysqld) executeMysqlCommands(connParams *sqldb.ConnParams, sql string) error {
-	return mysqld.executeMysqlScript(connParams, strings.NewReader(sql))
+	params, err := dbconfigs.WithCredentials(connParams)
```

Comment:
I think that's independent, actually. We only use the root user in two situations:

1. we have just created an empty database from scratch, it has no users, and we're about to run the init sql script. Since there is no user configured, we have to use root without password.

2. we are restoring from backup, and we want to run mysql_upgrade. In that case, we run mysql with --skip-grant-tables and --skip-networking, so we don't need the password, but the database is not visible from the outside either.

In both these cases, your uses root password is not needed at all. Does that make sense?

We never use the root user otherwise, as a hardcoded 'root' user. Of course, you can configure your dba user name to be root (instead of vt_dba) and password to be set, and we'll use that when doing DBA tasks.